testing/README.md

```diff
@@ -0,0 +1,95 @@
+# How to Test E2E Resource Changes
+This guide will give a step by step instruction on how to test changes made to E2E testing resources before pushing a PR. 
+
+### 1. Create an IAM Role with OIDC Identity Provider
+This step is needed to allow Github Action to have access to resources in the AWS account
+#### Create an OIDC Provider
+- First step is to create an OIDC Identity Provider to allow Github action access to the AWS account resource. Login to AWS, go to the IAM console and click on the Identity Providers tab.
+- Click on Add Provider, choose OpenID Connect and type `https://token.actions.githubusercontent.com` in the Provider URL. Click "Get thumbprint". For Audience, use `sts.amazonaws.com`. Finally, click "Add provider"
+#### Create an IAM role
+- Next, an IAM role needs to be created using the OIDC Identity Provider. Go to the Roles tab and click Create role. 
+- Choose Web Identity, and choose `token.actions.githubusercontent.com` as the Identity provider, Audience as `sts.amazonaws.com`, and for Github organizations put your github username down. Click next.
+- Add the AdministratorAccess policy. Click next.
+- Enter your Role name. Click "Create role".
+#### Add Additional Permission
+- After the role is created, search the role name in the roles tab, click on the role, and go to the Trust relationships tab. Click on "Edit trust policy".
+- In the Statement list, add the following item: 
+`{
+  "Sid": "accessToRole",
+  "Effect": "Allow",
+  "Principal": {
+  "AWS": "arn:aws:iam::<AccountID>:root"
+  },
+  "Action": "sts:AssumeRole"
+  }`. This additional permission is need to allow Github Action to assume roles and have access to the EKS cluster. 
+
+Additional Resource: https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-amazon-web-services
+
+### 2. Create an EKS Cluster 
+The E2E EKS test uses an EKS cluster to deploy the sample apps.
+#### Setup Environment with the Appropriate Roles and Permissions.
+- First, assume Admin role from the test account by running `ada credentials update --account=<AccountID> --role=Admin --provider=isengard --once`
+- Assume the e2e test role by running 
+  - `output=$(aws sts assume-role --role-arn arn:aws:iam::<AccountID>:role/<E2ETestRole> --role-session-name AWSCLI-Session)`
+  - `export AWS_ACCESS_KEY_ID=$(echo $output | jq -r .Credentials.AccessKeyId)`
+  - `export AWS_SECRET_ACCESS_KEY=$(echo $output | jq -r .Credentials.SecretAccessKey)`
+  - `export AWS_SESSION_TOKEN=$(echo $output | jq -r .Credentials.SessionToken)`
+- Run `aws sts get-caller-identity` to check if you are in the correct role
+#### Create a new Cluster
+- Next, create the cluster by running `eksctl create cluster --name <ClusterName> --region us-east-1 --zones us-east-1a,us-east-1b`. This will take around ~10 minutes. 
```

Comment:
Always point out the reader needs to replace something in the command before the command itself. Room for error here

Reply:
Done